modules/search/search.go

```diff
@@ -0,0 +1,118 @@
+package search
+
+import (
+	"github.com/bwmarrin/discordgo"
+	"github.com/lordralex/absol/api"
+	"github.com/lordralex/absol/api/database"
+	"github.com/lordralex/absol/api/logger"
+	"github.com/lordralex/absol/modules/factoids"
+	"github.com/spf13/viper"
+	"strconv"
+	"strings"
+)
+
+type Module struct {
+	api.Module
+}
+
+// Load absol commands API
+func (*Module) Load(ds *discordgo.Session) {
+	api.RegisterCommand("search", RunCommand)
+
+	api.RegisterIntentNeed(discordgo.IntentsGuildMessages, discordgo.IntentsDirectMessages)
+}
+
+func RunCommand(ds *discordgo.Session, mc *discordgo.MessageCreate, _ string, args []string) {
+	if mc.GuildID != "" {
+		_ = factoids.SendWithSelfDelete(ds, mc.ChannelID, "This command may only be used in DMs.")
+		return
+	}
+
+	if len(args) == 0 {
+		_, _ = ds.ChannelMessageSend(mc.ChannelID, "You must specify a search string!")
+		return
+	} else if len(strings.Join(args, "")) < 3 {
+		_, _ = ds.ChannelMessageSend(mc.ChannelID, "Your search is too short!")
+		return
+	}
+
+	max := viper.GetInt("factoids.max")
+	if max == 0 {
+		max = 5
+	}
+
+	db, err := database.Get()
+	if err != nil {
+		_, _ = ds.ChannelMessageSend(mc.ChannelID, "Failed to connect to database")
+		logger.Err().Printf("Failed to connect to database\n%s", err)
+		return
+	}
+
+	pageNumber := 0
+	pageNumber, err = strconv.Atoi(args[len(args)-1]) // if the last arg is a number use it as the page number
+
+	// if the page number was specified then we subtract one from it to make the page index start at 1, then
+	// cut the last argument out if it's a number
+	if _, err := strconv.Atoi(args[len(args)-1]); err == nil {
+		pageNumber = pageNumber - 1
+		args = args[:len(args)-1]
+	}
+
+	message := ""
+	// searches through results for a match
+	// gets the factoids table
+	var factoidsList []factoids.Factoid
+	var rows int64
+	db.Where("content LIKE ? OR name LIKE ?", "%"+strings.Join(args, " ")+"%", "%"+strings.Join(args, " ")+"%").Offset(pageNumber * max).Limit(max).Find(&factoidsList)
+	db.Where("content LIKE ? OR name LIKE ?", "%"+strings.Join(args, " ")+"%", "%"+strings.Join(args, " ")+"%").Table("factoids").Count(&rows) // this is a different query anyway so it just needs to be a seperate line to get the total number of results
+
+	// ensures that page number is valid
+	if pageNumber < 0 || pageNumber > int(rows)/max+1 {
```

Comment:
Just puke the last page, we already did heavy work on this

Reply:
I don't know how this would be done, as it would rely on knowing the count before the find, and that's not possible without doing the aforementioned heavy computing

Reply:
It's very simple to just throw the *first* page back at them, i'll add something to do that